### ISSUE TITLE
HW4 Idea

### ISSUE THREAD

**Issue by amah001** [mikeizbicki/ucr-cs100#1502]
Eduardo Carillo and I would like to make video tutorials for gdb with the possibility of doing it for strace as well. We would also rework their written tutorials to be more in depth.

**Comment by mikeizbicki** [comment]
There are already some videos on gdb: https://github.com/mikeizbicki/ucr-cs100/blob/2015spring/videos/GDB.md

I think there's a lot of room for improvement on these videos, however, so I'm open to approving that topic if you can convince me your videos will be better somehow.  I also think strace would be a good idea.

If you do want to do this topic, I'll want to speak with you both in person before I actually approve it.

**Comment by ecarr007** [comment]
We will start off by showing how to start using gdb by adding the -g flag, and showing what the simple functions are such as run, print, and next. We will then demonstrate how to use them using a program similar to the one shown in lab 3. We can then make a video showing the more advanced functions and a video showing us going in blind into a program with bugs and using gdb to help debug it.

For Strace, we will do basically the same concept using strace on a program and explaining the out put. We will then use the -e flag and explain what it does and how it is useful. We will do this for all the strace flags such as -o and -r. 

It will also be helpful to explain which debugging tool to use in which cases, such as using gdb to find segfaults, or strace to determine the system calls being run.

**Comment by mikeizbicki** [comment]
Sounds good.  Some comments:
1.  I'd like the gdb video to present the commands in the same order as the lab if at all possible.  This might mean adjusting the lab a bit, I'm not sure.
2. I'd like to see your strace video use the example I used in class: i.e. we didn't know how ls was collecting the information about the specific files, so we ran it with strace and saw the `stat` syscall being used.
